crates/cxx-qt-lib/src/gui/qimage.rs

```diff
@@ -332,6 +371,73 @@ impl QImage {
             None
         }
     }
+
+    /// Constructs a QImage from an existing memory buffer.
+    ///
+    /// # Safety
+    /// For details on safety see the [Qt documentation](https://doc.qt.io/qt-6/qimage.html#QImage-7)
+    pub unsafe fn from_raw_parts(
+        data: *const ffi::uchar,
+        width: i32,
+        height: i32,
+        format: QImageFormat,
+        image_cleanup_function: extern "C" fn(x: *mut ffi::c_void),
+        image_cleanup_data: *mut ffi::c_void,
+    ) -> Self {
+        ffi::qimage_init_from_raw_parts(
+            data,
+            width,
+            height,
+            format,
+            QImageCleanupFunction(image_cleanup_function),
+            image_cleanup_data,
+        )
+    }
+
+    /// Constructs a QImage from an existing mutable memory buffer.
+    ///
+    /// # Safety
+    /// For details on safety see the [Qt documentation](https://doc.qt.io/qt-6/qimage.html#QImage-8)
+    pub unsafe fn from_raw_parts_mut(
+        data: *mut ffi::uchar,
+        width: i32,
+        height: i32,
+        format: QImageFormat,
+        image_cleanup_function: extern "C" fn(x: *mut ffi::c_void),
+        image_cleanup_data: *mut ffi::c_void,
+    ) -> Self {
+        ffi::qimage_init_from_raw_parts_mut(
+            data,
+            width,
+            height,
+            format,
+            QImageCleanupFunction(image_cleanup_function),
+            image_cleanup_data,
+        )
+    }
+
+    /// Constructs a QImage from the raw data inside a `Vec<u8>`.
+    ///
+    /// # Safety
+    /// This function is unsafe because it requires that the data matches the given QImageFormat,
+    /// width and height.
```

Comment:
and that the data lives long enough?

Reply:
This isn't necessary here. As this function takes a `Vec<u8>` the QImage gets ownership of the data.

Reply:
:+1: